### PR TITLE
Fix Issue 21936 - Segfault with -dip1000

### PR DESCRIPTION
**Pull request by UplinkCoder**

### CODE REVIEW

File: test/compilable/issue21936.d [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// REQUIRED_ARGS: -preview=dip1000
+module issue21212;
+
+struct Handlers {
+    void registerType(T)()
+    {
+        registerConstructors!T;
+    }    void registerConstructors(T)()
```
Comment (thewilsonator) [inline test/compilable/issue21936.d:8]: spacing

File: test/compilable/issue21936.d [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        constructImplicit!T;
+    }
+}
+
+auto constructImplicit(T)(T.init.tupleof)
+{
+}
+
+import std.datetime;
```
Comment (thewilsonator) [inline test/compilable/issue21936.d:18]: NO

**Review by thewilsonator — CHANGES_REQUESTED** [review]

**Comment by UplinkCoder** [comment]
@thewilsonator valid points. fixed now.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
If this is a regression then this should target stable.

**Comment by UplinkCoder** [comment]
@thewilsonator I couldn't figure out how to make it target stable.
can you do this?

**Comment by thewilsonator** [comment]
follow the bot's instructions

**Comment by UplinkCoder** [comment]
@thewilsonator had to force push because there was still a std.datetime import

**Pull request merged**